cmake: Add option to specify custom zoneinfo

lat9nq opened the pull request:
Enables developers to use a different zoneinfo source than the included submodule by specifying the option `TZDB2NX_ZONEINFO_DIR`.

Also a couple of small fixes.

Closes #16 
@tachi107 please check if this is what you were asking for.

lat9nq committed:
cmake: Check program existence with CMake standards

lat9nq committed:
cmake: Use git clone instead of file copy

Preserves (enough) git repository data for configuration to work without error.

lat9nq committed:
cmake: Add option to specify custom zoneinfo

Enables developers to use a different zoneinfo source than the included submodule.

lat9nq commented:
Example usage would be `cmake .. -DTZDB2NX_ZONEINFO_DIR=/usr/share/zoneinfo/posix`

Tachi107 commented on externals/tz/CMakeLists.txt:
```diff
@@ -1,43 +1,58 @@
-    else()
-        set(TZ_MAKEFLAGS)
-    endif()
+    if (NOT TZDB2NX_ZONEINFO_DIR)
```
Maybe it'd be better to use `NOT EXISTS`?

lat9nq replied:
No, I specifically want to compare the string value of the variable. I'll add a comment there.

Tachi107 commented:
Thanks! This is _almost_ what I tried to do myself. The only issue with this patch is that having the tzdb submodule is still required to get the value of the `TZDB_VERSION` variable. Maybe we could use the `/usr/share/zoneinfo/tzdata.zi` file to read the version, since on my system the first line of the file contains `# version 2023d`

lat9nq commented:
> Thanks! This is _almost_ what I tried to do myself. The only issue with this patch is that having the tzdb submodule is still required to get the value of the `TZDB_VERSION` variable. Maybe we could use the `/usr/share/zoneinfo/tzdata.zi` file to read the version, since on my system the first line of the file contains `# version 2023d`

Thanks for looking over it. Not immediately sure how to handle the version. At the moment on my system I can tell it to go to `/usr/share/zoneinfo/posix` and that sub-directory will work sufficiently to produce the archive data, but there is no explicit version info there. I think perhaps the version would need to be specified via command line as well.

lat9nq committed:
cmake: Add user specifiable version variable

Previously it was still using the submodule's version.

lat9nq committed:
cmake: Add comment to directory check

In review, this if statement was questioned as being a directory existence check in fact it is checking the value of the variable itself.

lat9nq commented:
`TZDB2NX_VERSION` is a new variable you need to specify, now, if you wish to use `TZDB2NX_ZONEINFO_DIR`, and vice-versa.

Tachi107 commented:
I had another look at how to handle the version variable, and looking at tzdb's Makefile I saw that there are two targets that could be useful here.

The first one is the [`version`](https://github.com/eggert/tz/blob/8459570715214271fd5bab92ec4d6643901c02e0/Makefile#L665) target, which generates a `version` file containing the correct tzdb version. Its logic its a bit more advanced than the one used by your CMake scripts, so maybe it'd make sense to directly use that instead when building the submodule?

The second one is the [`tzdata.zi`](https://github.com/eggert/tz/blob/8459570715214271fd5bab92ec4d6643901c02e0/Makefile#L688), generating a file with the same name, which contains various information including the tzdb version as generated by the `version` target. This file is included in the `tzdata` package shipped by Debian, Ubuntu, Arch, and Fedora, and I think it'd be nice if the CMake scripts would read the version from there instead of requiring the user to define it themselves. The file is always located at `tzdir/tzdata.zi` (which in practice is almost always `/usr/share/zoneinfo/tzdata.zi`)

> At the moment on my system I can tell it to go to `/usr/share/zoneinfo/posix` and that sub-directory will work sufficiently to produce the archive data, but there is no explicit version info there.

Why `/usr/share/zoneinfo/posix` instead of simply `/usr/share/zoneinfo`?

lat9nq commented:
I should clarify that the version text file that is normally created by this repository uses a different version format than the tz submodule has. Specifically, Nintendo uses whatever YYMMDD that they built the data. I use the date of the commit in the tz submodule.

So it follows that it makes more sense to require developers to manually specify this variable here, since it was never supposed to be the version numbers from eggert/tz anyway.

In testing I quickly realized that /usr/share/zoneinfo includes about 3 times more files than one of the major subdirectories, which is why I happened to pick posix here since it more closely follows the normal timezone data files. There wasn't any error it just took a little longer to build.

lat9nq commented:
> So it follows that it makes more sense to require developers to manually specify this variable here, since it was never supposed to be the version numbers from eggert/tz anyway.

I wouldn't expect any Switch guests to care if the version number suddenly was a different format (i.e. 2023d), but I also haven't done much of any research on this. Take that with a grain of salt.

lat9nq committed:
cmake: Allow specifying TZDB2NX_VERSION without other requirements

I was worried the version and custom zoneinfo dir were tied together, but it's fine. We can override the version whenever we want.

Tachi107 commented:
> So it follows that it makes more sense to require developers to manually specify this variable here, since it was never supposed to be the version numbers from eggert/tz anyway.

Makes sense, looks good to me then! Thanks :)

lat9nq commented:
Apologies that I forgot to merge this.

Tachi107 commented:
Thanks again :D
